### PR TITLE
Refactor Users

### LINKED ISSUE #75: Refactor Users
- [x] Rename Files
- [x] Remove Try & Catch from controllers
- [x] Use asyncMiddleware in routes

### PR DESCRIPTION
**Pull request by AmranElmasri**
relats #75

### CODE REVIEW

File: server/src/controllers/user/login.ts [hunk 1]
```diff
@@ -2,32 +2,21 @@ import { Request, Response, NextFunction } from 'express';
...
+    next(new Error());
+  }
+  const { password: hashedPassword, id, username } = rows[0];
+  const isMatched = compare(password, hashedPassword);
+  if (isMatched) {
+    const token = await signToken({ id, username });
+    res.cookie('token', token, { httpOnly: true, secure: false })
+      .status(200).json({ message: 'login successfully!' });
+  } else {
+    next(new Error());
```
Comment (Nizar7zak) [inline server/src/controllers/user/login.ts]: here, you can send a response to the client side,
return res.status(4xx).json({message:' something'})

File: server/src/controllers/user/signUp.ts [hunk 2]
```diff
@@ -2,39 +2,28 @@ import { Request, Response, NextFunction } from 'express';
...
+      username,
+      password: encryptedPass,
+      email,
+    });
+    const { id } = rows[0];
+    const token = signToken({ id, username });
+    res
+      .cookie('token', token)
+      .status(200)
+      .json({ msg: ' Account created successfully' });
```
Comment (Nizar7zak) [inline server/src/controllers/user/signUp.ts]: be consistent please, we send a message in json object not msg

File: server/src/controllers/user/signUp.ts [hunk 3]
```diff
@@ -2,39 +2,28 @@ import { Request, Response, NextFunction } from 'express';
...
+  } else {
+    next(new Error());
```
Comment (Nizar7zak) [inline server/src/controllers/user/signUp.ts]: what the error here?

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

File: server/src/controllers/user/signUp.ts (hunk 3 above)
Comment (AmranElmasri) [reply server/src/controllers/user/signUp.ts]: Invalid verification code

File: server/src/routes/index.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import express from 'express';
 import tasksRoute from './tasksRoute';
 import projectsRoute from './projectsRoute';
 import sectionsRoute from './sectionsRoute';
-import userRoute from './userRoute';
+import userRoute from './usersRoute';
 import { sendEmailController } from '../controllers';
```
Comment (nada-abuzaid) [inline server/src/routes/index.ts]: Remove sendEmailController because we move it to user routes.

File: server/src/controllers/user/sendEmail.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Request, Response } from 'express';
+import { sendCode, sendEmailSchema } from '../../utils';
+
+const sendEmail = async (req: Request, res: Response) => {
+  const { email } = req.body;
+  await sendEmailSchema.validateAsync({ email });
+
+  await sendCode(email);
+
```
Comment (nada-abuzaid) [inline server/src/controllers/user/sendEmail.ts]: please remove the white spaces here (Line 7 & 9)

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]
Great Amran!

**Review by nada-abuzaid — APPROVED** [review]

**Review by karamzomlot — APPROVED** [review]

**Review by MaramMs — APPROVED** [review]

**Pull request merged**